test/lint.js

```diff
@@ -0,0 +1,145 @@
+'use strict';
+
+let assert = require('assert');
+let emu = require('../lib/ecmarkup');
+
+function alg([before, after], marker) {
+  return `<emu-alg>${before}${marker}${after}</emu-alg>`;
+}
+
+let M = '@@@'; // Marks the location of the error
+
+describe('linting', function () {
+  describe('grammar sanity', function () {
+    it('unused parameters', async function () {
+      await assertLint(
+        `
+          <emu-grammar type="definition">
+            Statement[${M}a]: \`;\`
+          </emu-grammar>
+        `,
+        "Parameter 'a' is unused."
+      );
+    });
+
+    it('missing parameters', async function () {
+      await assertLint(
+        `
+          <emu-grammar type="definition">
+            Foo[a]:
+              [+a] \`a\`
+              \`b\`
+
+            Bar: ${M}Foo
+          </emu-grammar>
+        `,
+        "There is no argument given for parameter 'a'."
+      );
+    });
+  });
+
+  describe('grammar+SDO sanity', function () {
+    it('undefined nonterminals in SDOs', async function () {
+      await assertLint(
+        `
+          <emu-grammar>
+            ${M}Statement: EmptyStatement
+          </emu-grammar>
+          <emu-alg>
+            1. Return Foo.
+          </emu-alg>
+        `,
+        'Could not find a definition for LHS in syntax-directed operation'
+      );
+    });
+
+    it('undefined productions in SDOs', async function () {
+      await assertLint(
+        `
+          <emu-grammar type="definition">
+              Statement: \`;\`
+          </emu-grammar>
+
+          <p>SS: Foo</p>
+          <emu-grammar>
+            Statement: ${M}EmptyStatement
+          </emu-grammar>
+          <emu-alg>
+            1. Return Foo.
+          </emu-alg>
+        `,
+        'Could not find a production matching RHS in syntax-directed operation'
+      );
+    });
+  });
+
+  describe('algorithm line endings', function () {
```

Comment:
This doesn't even come close to covering this rule. Also, I'd probably split it into its own file. And no negative tests?

Reply:
> This doesn't even come close to covering this rule.

Complete tests for this linting rule did not seem necessary to me. Feel free to push up a commit with some, if you want.

> Also, I'd probably split it into its own file.

... Why? That might make sense once we get more tests, I guess, but I don't see the point.

> And no negative tests?

Added a negative test.

Reply:
I'm okay with not getting 100% coverage for now, but would like to get to that eventually. We should still split it out of this file though, as we expect to add more linting rules and don't want them to continuously grow this file.

Reply:
> We should still split it out of this file

Done.